pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1843,7 +1843,10 @@ class Timedelta(_Timedelta):
                              NPY_DATETIMEUNIT.NPY_FR_W,
                              NPY_DATETIMEUNIT.NPY_FR_GENERIC]):
                 err = npy_unit_to_abbrev(reso)
-                raise ValueError(f" cannot construct a Timedelta from a unit {err}")
+                raise ValueError(
+                    "Units 'M', 'Y', and 'y' are no longer supported, as they do not "
```

Comment:
A message like `f"Unit {err} is not supported. Allowed units are ..."` might be more general. A user here might pass a np.timedelta object that is higher resolution than ns for example

Reply:
that's a good point, thanks, numpy allows picoseconds and other very small things

Reply:
Updated with suggestions but still kept the reference to ambiguous time periods being no longer supported.